azure/services/privatedns/privatedns.go

```diff
@@ -57,15 +59,51 @@ func (s *Service) Reconcile(ctx context.Context) error {
 
 	zoneSpec := s.Scope.PrivateDNSSpec()
 	if zoneSpec != nil {
+		// Skip the reconciliation of private DNS zone which is not managed by capz.
+		isManaged, err := s.isPrivateDNSManaged(ctx, s.Scope.ResourceGroup(), zoneSpec.ZoneName)
+		if err != nil && !azure.ResourceNotFound(err) {
+			return errors.Wrapf(err, "could not get private DNS zone state of %s in resource group %s", zoneSpec.ZoneName, s.Scope.ResourceGroup())
+		}
+		// If resource is not found, it means it should be created and hence setting isVnetLinkManaged to true
+		// will allow the reconciliation to continue
+		if err != nil && azure.ResourceNotFound(err) {
+			isManaged = true
+		}
+		if !isManaged {
+			s.Scope.V(2).Info("Skipping reconciliation of unmanaged private DNS zone", "private DNS", zoneSpec.ZoneName)
+			return nil
+		}
 		// Create the private DNS zone.
 		s.Scope.V(2).Info("creating private DNS zone", "private dns zone", zoneSpec.ZoneName)
-		err := s.client.CreateOrUpdateZone(ctx, s.Scope.ResourceGroup(), zoneSpec.ZoneName, privatedns.PrivateZone{Location: to.StringPtr(azure.Global)})
+		pDNS := privatedns.PrivateZone{
```

Comment:
Maybe not part of this pr, but should we call `CreateOrUpdateZone` if there is already a zone existing since we don't allow changes to any of its properties?

Reply:
Opened up an issue to keep track of this one. 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1872